smarts/sstudio/types.py

```diff
@@ -890,6 +936,12 @@ def __post_init__(self):
                 "Only boids can have keep_alive enabled (for persistent boids)"
             )
 
+        poly = None
+        if not isinstance(self.zone, MapZone):
+            poly = self.zone.to_geometry(road_map=None)
+        if poly is not None and (poly.is_valid == False):
+            raise ValueError("The Zone Polygon is not a valid closed loop")
```

Comment:
What does it mean if it is `None`? Is that check needed? In any case, this could be simplified:

```python
if not isinstance(self.zone, MapZone):
    poly = self.zone.to_geometry(road_map=None)
    if poly is not None and not poly.is_valid:
        raise ValueError("The Zone Polygon is not a valid closed loop")
```

Reply:
I think you are right, the check for poly is None is not needed, will use the simplified version, thanks!